### DISCUSSION TITLE
trying with htmgo

### DISCUSSION THREAD

**Discussion by rowinf** [pocketbase/pocketbase#5886] [Q&A] [answered]
I made an attempt with pocket base and htmgo and it didn't seem to work. 

I was following the "extending with go" example in the docs on a project generated with htmgo. 

Would anyone be able to give me a hint in the right direction? 

my code: https://github.com/rowinf/htmgo-pocketbase

**Comment by ganigeorgiev** [comment]
What is `h.Start`, does it start its own listener?

I have very little experience with htmx (and none with htmgo) so I'm not sure I can help you here but note that the PocketBase `app.Start()` initializes the default PoketBase commands and when that command is `serve` it is a blocking/long-running task (aka. the web server) so anything after it will not run and need to be either attached via hook or executed in a separate goroutine.

**Comment by dschissler** [comment, marked as answer]
`htmgo uses std http with chi router as its web server, *h.RequestContext is a thin wrapper around *http.Request. A page must return *h.Page, and accept *h.RequestContext as a parameter`

I've looked at two or three of these htmx + go libraries to conclude that they need to be ported to have any usefulness.  There is one that sets up a controller object on each request by wrapping in the reader and writer, but otherwise you're just fighting against the help too much.
